.github/workflows/ci.yml

```diff
@@ -0,0 +1,138 @@
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+  ci:
+    strategy:
+      # Allows for matrix sub-jobs to fail without canceling the rest
+      fail-fast: false
+
+      # MATRIX:
+      # =======
+      # Required parameters:
+      #  os                  the os to run on
+      #  python-version      the python version to use
+      #  backend             the backend to use
+      #  env                 any additional env variables. Set to '{}' for none
+      # Optional parameters:
+      #  extra_hash          extra hash str to differentiate from other caches with similar name (must always start with '-')
+      matrix:
+        # Tests [amd64]
+        #
+        # FIXME: 'cpp' tests seems to fail due to compilation errors (numpy_pythran_unit)
+        # in all python versions and test failures (builtin_float) in 3.5<
+        os: [ubuntu-18.04]
+        backend: [c, cpp]
+        python-version: [2.7, 3.4, 3.5, 3.6, 3.7, 3.8, 3.9, 3.10-dev]
+        env: [{}]
+
+        include:
+          # Ubuntu sub-jobs:
+          # ================
+          # Linting
+          - os: ubuntu-18.04
+            python-version: 3.7
+            backend: "c,cpp"
+            env: { TEST_CODE_STYLE: 1 }
+            extra_hash: "-codestyle"
+          # Limited API
+          - os: ubuntu-18.04
+            python-version: 3.6
+            backend: "c,cpp"
+            env: { LIMITED_API: "--limited-api", EXCLUDE: "--no-file" }
+            extra_hash: "-limited_api"
+          - os: ubuntu-18.04
+            python-version: 3.7
+            backend: "c,cpp"
+            env: { LIMITED_API: "--limited-api", EXCLUDE: "--no-file" }
+            extra_hash: "-limited_api"
+          - os: ubuntu-18.04
+            python-version: 3.8
+            backend: "c,cpp"
+            env: { LIMITED_API: "--limited-api", EXCLUDE: "--no-file" }
+            extra_hash: "-limited_api"
+          # Stackless
+          - os: ubuntu-18.04
+            python-version: 2.7
+            backend: c
+            env: { STACKLESS: true, PY: 2 }
+            extra_hash: "-stackless"
+          - os: ubuntu-18.04
+            python-version: 3.6
+            backend: c
+            env: { STACKLESS: true, PY: 3 }
+            extra_hash: "-stackless"
+          # Pypy
+          - os: ubuntu-18.04
+            python-version: pypy-2.7
+            backend: c
+            env: {}
+          - os: ubuntu-18.04
+            python-version: pypy-3.6
```

Comment:
Please use pypy-3.7, PyPy is dropping support for 3.6

Reply:
Is there an easy way to just get the most recent release of pypy-3? This seems closest to what Cython practically tries to support.

Reply:
Unfortunately it seems the [github action for python-versions](https://github.com/actions/setup-python#available-versions-of-pypy) does not support that, they [link pypy3 to pypy-3.6](https://github.com/actions/setup-python/pull/166) for backward compatibility.

Reply:
> 
> 
> Please use pypy-3.7, PyPy is dropping support for 3.6

That is up to @scoder. CPython2 is still supported by Cython even tho there is no more support

Reply:
I meant changing `python-version: pypy-3.6` to `python-version: pypy-3.7` here and a few lines down

Reply:
Yes, I know what you meant, but I'm not sure if scoder wants to keep it as pypy-3.6

Reply:
Ahh, ok, I was confused by the mention of CPython2. 

I guess the question is if there is really a community of PyPy3.6 users who should be preferred over PyPy3.7 users. It doesn't make sense to spend too many precious CI resources to test two versions of PyPy.

My justification (for @scoder's consideration) goes something like: any consumers of Cython on PyPy are third-party users installing a package that is kind enough to build for them. People who build their own code will use something like cffi instead. So those third-party consumers of other packages may not even know Cython underlies their code. Since the packages they consume are dropping Python3.6, it makes sense to test for what should be the more up-to-date PyPy3.7. PyPy3.7 will also be the only one to fix bugs going forward (until PyPy3.8 is released).

Reply:
Let's try to always test against the latest PyPy(3) releases. PyPy fixes issues all the time and their average user probably won't depend on a "system provided PyPy distribution", so supporting anything but the latest is just duplicating their work on our side.

Reply:
So I update it to pypy3.7?

Reply:
Please do.